validations/users.js

```diff
@@ -7,24 +10,40 @@ const userRegValidators = [
         .isLength({ max: 50 })
         .withMessage('User Name must not be more than 50 characters long')
         .matches(/@/)  
-        .withMessage('User Name cannot contain @ symbol'),
+        .withMessage('User Name cannot contain @ symbol')
+        .custom(async (value) => {
+            const username = await User.findOne({where: {username: value}})
+            if(!username) {
+                throw new Error('User Name already in use. Please choose another.')
+            }
+        }),
     check('email')
         .exists({ checkFalsy: true })
         .withMessage('Please provide a value for Email')
         .isLength({ max: 50 })
         .withMessage('Email must not be more than 50 characters long')
         .isEmail()
-        .withMessage('Email is not a valid email'),
+        .withMessage('Email is not a valid email')
+        .custom(async (value) => {
+            const email = await User.findOne({where: {email: value}})
+            if(!email) {
+                throw new Error('Email already in use. Please sign-in or choose another email to register.')
+            }
+        }),
     check('password')
+        .trill()
         .exists({ checkFalsy: true })
         .withMessage('Please provide a value for Password')
-        .isLength({ max: 50 })
-        .withMessage('Password must not be more than 50 characters long'),
+        .isLength({ min: 8, max: 50 })
+        .withMessage('Password must be at least 8 characters long and not more than 50 characters long')
+        .matches(/(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*\W)/)
+        .withMessage('Password must contain at least one capital letter, one lower case letter, one number and one symbol'),
     check('confirmPassword') 
+        .trill()
         .exists({ checkFalsy: true })
         .withMessage('Please provide a value for Confirm Password')
-        .isLength({ max: 50 })
-        .withMessage('Confirm Password must not be more than 50 characters long'),
+        .matches('password')

```

Comment:
What is here will just check if the confirm matches the string of "password". What you want is this:
```
.custom((value, { req }) => {
      if (value !== req.body.password) {
        throw new Error('Confirm Password does not match Password');
      }
      return true;
    }